pkg/virt-controller/watch/vmi/vmi.go

```diff
@@ -1102,7 +1104,21 @@ func (c *Controller) handleBackendStorage(vmi *virtv1.VirtualMachineInstance) (s
 		return "", nil
 	}
 
-	pvc := backendstorage.PVCForVMI(c.pvcIndexer, vmi)
+	// If we found a successful migration, critical PVC labels were updated, we need to refresh the informer
+	c.pvcExpectations.ExpectCreations(key, 1)
+	pvc, err, success := backendstorage.RecoverFromBrokenMigration(c.clientset, c.migrationIndexer, c.pvcIndexer, vmi, c.templateService.GetLauncherImage())
```

Comment:
Why do we run the recovery check on the vmi controller part and not from the VM?

Reply:
I guess I was assuming that doing it in the VMI controller would cover both VMs and VMIs, would it not?
Do you think it should be done in both the VM and VMI controllers, like in https://github.com/kubevirt/kubevirt/pull/13636?
Maybe we can defer it to that PR? Let me know what you think

Reply:
When I tried with volume migration, the VMI controller wasn't reconciliated and then the recovery wasn't triggered. However, since the recovery is supposed to happen when the VM is (re-)started, don't you think it is more consistent in doing it in the VM controller in the `startVMI`?

Reply:
That's correct, recovery only makes sense for VMs, not VMIs, will move, thanks!